modules/system/defaults/dock.nix

```diff
@@ -140,13 +140,13 @@ in {
       type = types.nullOr (types.listOf (types.either types.path types.str));
       default = null;
       example = [ "~/Documents" "~/Downloads" ];
-      description = lib.mdDoc ''
+      description = ''
         Persistent folders in the dock.
       '';
       apply = value:
         if !(isList value)
         then value
-        else map (folder: { tile-data = { file-data = { _CFURLString = folder; _CFURLStringType = 15; }; }; }) value;
+        else map (folder: { tile-data = { file-data = { _CFURLString = "file://" + folder; _CFURLStringType = 15; }; }; tile-type = if strings.hasInfix "." folder then "file-tile" else "directory-tile"; }) value;
```

Comment:
Folders can have `.` in their name like `~/.config`, ideally we should split by `/` then just look at the last part

Reply:
You're right, I've pushed the changes.